lib/cancan/conditions_matcher.rb

```diff
@@ -78,7 +78,8 @@ def condition_match?(attribute, value)
 
     def hash_condition_match?(attribute, value)
       if attribute.is_a?(Array) || (defined?(ActiveRecord) && attribute.is_a?(ActiveRecord::Relation))
-        attribute.any? { |element| matches_conditions_hash?(element, value) }
+        match_results = attribute.map { |element| matches_conditions_hash?(element, value) }
+        match_results.any?
```

Comment:
rather than the multi-line solution, how do you feel about `attribute.to_a.any? { |element| matches_conditions_hash?(element, value) }`

if `attribute` is already an array, `to_a` will do nothing. if it's not, it'll force the relation to be loaded.

i just think that code makes a bit more sense than what you have here - if i saw this, i'd want to refactor it back to the old version.

Reply:
That absolutely makes a lot of sense ^^ Your change looks a lot cleaner and thus I'm happy to revert to just adding to_a to force the relation loading